projects/components/src/back-button/back-button.component.test.ts

```diff
@@ -0,0 +1,33 @@
+import { IconType } from '@hypertrace/assets-library';
+import { NavigationService } from '@hypertrace/common';
+import { createHostFactory, mockProvider, Spectator } from '@ngneat/spectator/jest';
+import { MockComponent } from 'ng-mocks';
+import { ButtonRole } from '../button/button';
+import { ButtonComponent } from '../button/button.component';
+import { BackButtonComponent } from './back-button.component';
+
+describe('Back Button Component', () => {
+  let spectator: Spectator<BackButtonComponent>;
+
+  const createHost = createHostFactory({
+    component: BackButtonComponent,
+    declarations: [MockComponent(ButtonComponent)],
+    providers: [mockProvider(NavigationService)],
+    shallow: true
+  });
+
+  test('should render the button correctly', () => {
+    spectator = createHost(`<ht-back-button></ht-back-button>`);
+
+    expect(spectator.query(ButtonComponent)).toExist();
```

Comment:
nit: can query once before calling the expect methods

Reply:
Updated